tests/core/request.test.ts

```diff
@@ -44,6 +44,43 @@ describe('Request properties', () => {
 
       await fetch('/abc/def').expect(200, '/def')
     })
+    it('should set the correct req.url on routes even in a subapp', async () => {
+      const makeApp = () =>
+        new App().get('/a1/b/*', (req, res) => res.send(req.url)).get('/a2/b/:pat', (req, res) => res.send(req.url))
+
+      const app = makeApp()
+      app.use('/s/:pat1/:pat2', makeApp())
```

Comment:
Add a test case for when a subapp is mount on a wildcard route.

Reply:
Done in 7c3896fc4ac60aab28a92701622117b65482a08a, but note that since `*` is greedy the only possible paths for those subapps is `/`